### PR TITLE
Add project group models and migration

### PR DESCRIPTION
**Pull request by baopham**
Add Project Group models and setup many-to-many relationship with Project

> Note for development:
> To run test: `RAILS_ENV=test EAGER_LOAD=1 DEFAULT_URL=http://www.test-url.com forking-test-runner test/models/project_*.rb --merge-coverage --quiet`

/cc @timotheeg @ningt 

### Tasks
 - [ ] :+1: from team

### References
 - Jira link: n/a

### Risks
- Low: should not cause any regression.

### CODE REVIEW

**Comment by baopham** [comment]
we should setup Travis for this repo...

File: app/models/project.rb [hunk 1]
```diff
@@ -27,6 +28,8 @@ class Project < ActiveRecord::Base
   has_many :user_project_roles, dependent: :destroy
   has_many :users, through: :user_project_roles
   has_many :secret_sharing_grants, dependent: :destroy
+  has_many :projects_project_groups
```
Comment (ningt) [inline app/models/project.rb]: should we call it `project_group_projects`?
https://github.com/zopim/samson-playground/blob/master/app/models/deploy_group.rb#L10

Comment (baopham) [reply app/models/project.rb]: Usually it follows lexical order https://stackoverflow.com/questions/11590469/rails-naming-convention-for-join-table `'projects' < 'project_groups'`

File: app/models/project.rb [hunk 2]
```diff
@@ -27,6 +28,8 @@ class Project < ActiveRecord::Base
...
+  has_many :projects, through: :projects_project_groups
```
Comment (ningt) [inline app/models/project.rb]: should here be `has_many :project_groups`?

Comment (baopham) [reply app/models/project.rb]: yeah, good catch

**Pull request merged**